mongo/crud_examples_test.go

```diff
@@ -0,0 +1,525 @@
+
+	// specify a pipeline that will only match "insert" events
+	// specify the MaxAwaitTimeOption to have each attempt wait two seconds for new documents
+	matchStage := bson.D{{"$match", bson.D{{"operationType", "insert"}}}}
+	opts := options.ChangeStream().SetMaxAwaitTime(2 * time.Second)
+	changeStream, err := db.Watch(context.TODO(), mongo.Pipeline{matchStage}, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// print out all change stream events in the order they're received
+	// see the mongo.ChangeStream documentation for more examples of using change streams
+	for changeStream.Next(context.TODO()) {
+		fmt.Println(changeStream.Current)
+	}
+}
+
+// Collection examples
+
+func ExampleCollection_Aggregate() {
+	var coll *mongo.Collection
+
+	// specify a pipeline that will return the number of times each name appears in the collection
+	// specify the MaxTime option to limit the amount of time the operation can run on the server
+	groupStage := bson.D{
+		{"$group", bson.D{
+			{"_id", "$name"},
+			{"numTimes", bson.D{
+				{"$sum", 1},
+			}},
+		}},
+	}
+	opts := options.Aggregate().SetMaxTime(2 * time.Second)
+	cursor, err := coll.Aggregate(context.TODO(), mongo.Pipeline{groupStage}, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// get a list of all returned documents and print them out
+	// see the mongo.Cursor documentation for more examples of using cursors
+	var results []bson.M
+	if err = cursor.All(context.TODO(), &results); err != nil {
+		log.Fatal(err)
+	}
+	for _, result := range results {
+		fmt.Printf("name %v appears %v times\n", result["_id"], result["numTimes"])
+	}
+}
+
+func ExampleCollection_BulkWrite() {
+	var coll *mongo.Collection
+
+	// insert {name: "Alice"} and delete {name: "Bob"}
+	// set the Ordered option to false to allow both operations to happen even if one of them errors
+	models := []mongo.WriteModel{
+		mongo.NewInsertOneModel().SetDocument(bson.D{{"name", "Alice"}}),
+		mongo.NewDeleteOneModel().SetFilter(bson.D{{"name", "Bob"}}),
+	}
+	opts := options.BulkWrite().SetOrdered(false)
+	res, err := coll.BulkWrite(context.TODO(), models, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Printf("inserted %v and deleted %v documents\n", res.InsertedCount, res.DeletedCount)
+}
+
+func ExampleCollection_CountDocuments() {
+	var coll *mongo.Collection
+
+	// count the number of times the name "Bob" appears in the collection
+	// specify the MaxTime option to limit the amount of time the operation can run on the server
+	opts := options.Count().SetMaxTime(2 * time.Second)
+	count, err := coll.CountDocuments(context.TODO(), bson.D{{"name", "Bob"}}, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("name Bob appears in %v documents", count)
+}
+
+func ExampleCollection_DeleteMany() {
+	var coll *mongo.Collection
+
+	// delete all documents in which the "name" field is "Bob" or "bob"
+	// specify the Collation option to provide a collation that will ignore case for string comparisons
```

Comment:
I found that this example made me go check the docs so see how this collation ignores case.  Turns out it does because CaseLevel defaults to false, but perhaps for clarity we should add that explicitly for purposes of this example.  Or else demonstrate something other than case insensitivity

Reply:
I didn't realize that `CaseLevel` mattered when writing this example. Added an explicit `CaseLevel: false` to the struct.